src/install.js

```diff
@@ -49,6 +50,65 @@ questions.optional = [
 	},
 ];
 
+function checkSetupEnv() {
+	winston.info("checking env vars for setup info...");
+	let envNbbRe = /NBB_(?!DB_).*/
+	let envNbbDbRe = /NBB_DB_.*/
+	let envopts = {
+		"NBB_URL": "url",
+		"NBB_PORT": "port",
+		"NBB_SECRET": "secret", // only a "cookie" for web setup???
+		"NBB_ADMIN_USERNAME": "admin:username",
+		"NBB_ADMIN_PASSWORD": "admin:password",
+		"NBB_ADMIN_EMAIL": "admin:email",
+		"NBB_DB": "database",
+		"NBB_DB_HOST": "host",
+		"NBB_DB_PORT": "port",
+		"NBB_DB_USER": "username",
+		"NBB_DB_PASSWORD": "password",
+		"NBB_DB_NAME": "database",
+		"NBB_DB_SSL": "ssl",
+	};
+
+	let evars = nconf.env().get();
+	var setupVal = Object()
+
+	Object.keys(evars).map(v => {
```

Comment:
Honestly this looks a little over complicated. I don't think it's really helping by using a regex nconf etc here. I think it would be better to just do:

```
const setupVal = {
  'admin:username': process.env['NODEBB_ADMIN_USERNAME'],
  'port': process.env['NODEBB_PORT'],
  ...
};
```

Reply:
the reason I used this method is that it allows to only define/overwrite any values in `setupVal` *if the env var is actually set*, thereby making it possible to *override* possibly existing config keys (also working in case the calling order would be changed) and ignoring unset ones, without needing any logic or check per key...

Reply:
Well how about you make a few changes fire improvement and then I'll see how it looks from there:

- Object.entries instead of Object.keys
- Use .startsWith instead of regexs
- `const setupVal = {};`
- use process.env instead of nconf.env.get